packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -114,6 +117,12 @@ library LibProposing {
                 blobHash: 0, // to be initialized below
                 extraData: params.extraData,
                 depositsHash: keccak256(abi.encode(depositsProcessed)),
+                // TODO(Brecht): A bit limiting this is required to be the same address as
+                // msg.sender
+                // And fees will be collected on L2 so if a smart contract is used as the proposer
+                // it requires
+                // the same contract to be deployed on L2 at the same address owned by the same
+                // entity.
```

Comment:
Having the address as an option I think would still be useful, otherwise have to take into account L1/L2 differences. Also could limit the composability of the smart contract because have to be very careful now which contract is `msg.sender` when submitting blocks.

Reply:
I'm ok with being able to customize the coinbase. Maybe a seperate PR for it?

Reply:
See https://github.com/taikoxyz/taiko-mono/pull/15743